exercicios/para-casa/src/controllers/cozinhaController.js

```diff
@@ -0,0 +1,328 @@
+    const { email } = request.body;
+    const hashedPassword = bcrypt.hashSync(request.body.password, 10)
+    request.body.password = hashedPassword
+
+    //verifica se o email já está cadastrado
+    const emailExiste = await UserSchema.exists({
+        email: request.body.email
+    })
+    if (emailExiste) {
+        return response.status(409).send({
+            message: 'Esse email já foi cadastrado!',
+        })
+    }
+
+    // //verifica se o email possui @
+    // const verificaArroba = email.indexOf('@')
+    // if (verificaArroba == -1) {
+    //     return response.status(401).send({
+    //         message: 'O endereço de email está inválido. Caracter a menos: arroba [ @ ]',
+    //     })
+    // }
+
+    // //verifica se o email possui ponto após o ARROBA
+    // const verificaPonto = email.includes(".", verificaArroba);
+    // if (!verificaPonto) {
+    //     return response.status(401).send({
+    //         message: 'O endereço de email está inválido. Caracter a menos: ponto [ . ]'
+    //     })
+    // }
+
+    //verifica se o email possui caracter depois do PONTO + 2 de cima
+    const emailRegex = /\S+@\S+\.\S+/
+    if (!emailRegex.test(email)) {
+        return response.status(401).send({
+            message: 'Email inválido!',
+        })
+    }
+
+    try {
+        const novoUsuario = new UserSchema(request.body)
+        const salvarUsuario = await novoUsuario.save()
+        response.status(201).send({
+            message: 'Usuário cadastrado com sucesso!',
+            salvarUsuario,
+        })
+    } catch (err) {
+        response.status(500).send({
+            message: err.message,
+        })
+    }
+}
+
+const getAll = async (req, res) => {
+    UserSchema.find(function (err, users) {
+        if (err) {
+            res.status(500).send({
+                message: err.message
+            })
+        }
+        res.status(200).send(users)
+    })
+}
+
+const buscarTodasCozinhas = async (request, response) => {
+    try {
+
+        const cozinha = await CozinhaSchema.find()
+
+        if (cozinha.length > 1) {
+            return response.status(200).json({ Prezades: `Encontramos ${cozinha.length} cozinhas.`, cozinha })
+        } else if (cozinha.length == 1) {
+            return response.status(200).json({ Prezades: `Encontramos ${cozinha.length} cozinha.`, cozinha })
+        } else {
+            return response.status(404).json({ Prezades: `Nenhuma cozinha foi encontrada.` })
+        }
+
+    } catch (error) {
+        response.status(500).json({
+            message: error.message
+        })
+    }
+}
+
+const buscarCozinhaId = async (request, response) => {
+
+    const { id } = request.params
+
+    try {
+        //Regras de negócio: Não aceitará id com menos de 24 caracteres
+        if (id.length > 24) {
+            return response.status(401).json({
+                Alerta: `Id incorreto. Caracter a mais: ${id.length - 24}.`
+            })
+        } else if (id.length < 24) {
```

Comment:
aqui poderíamos dar uma pequena resumida, uma sugestão é trocar o `else if` só por `else`, já que, nesse caso, tudo o que entra no `if `é ` id.length > 24` o que entra no `else` é tudo o que for diferente dessa verificação, nesse caso ` id.length < 24`, faz sentido?
```suggestion
       else {
            return response.status(401).json({
                Alerta: `Id incorreto. Caracter a menos: ${24 - id.length}.`
            })
        }
```

Reply:
Fez sim prof, mas quando coloquei ele não retornou mais a cozinha se achasse, sempre retornava o response do else. depois eu olho e testo melhor. Obrigada pela sugestão!

Reply:
ahhhhh, acabei de olhar aqui seu código inteiro pra essa funçao e entendi o pq disso acontecer: a gente tem um if logo em seguida pra verificar se a cozinha existe ou nao, quando transformamos o `else if` pra `else`, o restante do código fica `unreachable` e aí nao faz a verificação. Diferente do que eu havia entendido, são três casos ao invés de dois: o de `> 24` o de `< 24` e o de `= 24`. Pode ignorar essa sugestão e parabéns pela linha de raciocínio 😄

Reply:
ééé, inicialmente eu tentei assim mas não retornava aí fiz esses dois ifs. Eu agradeço demais o cuidado em olhar meu código! Se tiver outra sugestão pode me mandar! eu amo novas formas , bj no corê